File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/abstractClass/abstractExtendsNonAbstract/Animal.java
package de.fraunhofer.iem.classesAndObjects.classes.abstractClass.abstractExtendsNonAbstract;

/**
 * Abstract class extends the non-abstract class
 *
 * @author Ranjith Krishnamurthy
 */
abstract public class Animal extends NoiseBox {
    private String DNA;

    public Animal(String DNA) {
        this.DNA = DNA;
    }

    @Override
    public abstract void makeNoise();

    public void displayDNA() {
        System.out.println(DNA);
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/abstractClass/abstractExtendsNonAbstract/Animal.kt
package de.fraunhofer.iem.classesAndObjects.classes.abstractClass.abstractExtendsNonAbstract

/**
 * Abstract class extends the non-abstract class
 *
 * @author Ranjith Krishnamurthy
 */
abstract class Animal(private val DNA: String) : NoiseBox() {
    abstract override fun makeNoise()
    fun displayDNA() {
        println(DNA)
    }
}